src/test/run-pass/impl-trait-in-bindings.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(impl_trait_in_bindings)]
+
+use std::fmt::Debug;
+
+const FOO: impl Debug + Clone + PartialEq<i32> = 42;
+
+static BAR: impl Debug + Clone + PartialEq<i32> = 42;
+
+trait Foo {
+    const FOO: impl Clone;
+}
+
+impl Foo for i32 {
+    const FOO: impl Clone = 42;
+}
+
+fn main() {
+    let foo: impl Debug + Clone + PartialEq<i32> = 42;
```

Comment:
Can you add a test showing what happens when putting an `: impl Trait` `const`,  `static`, and `let` in a generic function and a closure in a generic function, both which capture generic arguments and that don't capture generic arguments? The only ones which *should* be able to capture generic arguments are `let` bindings inside generic functions and closures in generic functions. The current logic looks reasonable to me, but I want to make sure that the generics-from-the-parent hack is working even through a closure parent, which is the child of a function parent.

Reply:
Could you give an example of this actually? I don't see any way a let binding can use a type parameter in a non-trivial way in a function. Same with a closure.

Reply:
@cramertj But the impl type doesn't make use of the type parameter there...

Reply:
@alexreg what do you mean? It looks like it uses `T` to me.

Reply:
@cramertj I see what you mean... I meant in the declaration though. Something like `impl PartialEq<T>`.